NodeMCU/HWHandling.ino

```diff
@@ -0,0 +1,79 @@
+#include <ArduinoJson.h>
+#include <WebSocketsClient.h>
+#include <ESP8266WiFi.h>
+
+
+WebSocketsClient NodeWSocket;
+
+#define SERVER "wss://iot-chat-mhashim6.herokuapp.com/"
+#define PORT 80
+#define URL "/"
+
+DynamicJsonDocument doc(200);
+
+void websocketEvent(WStype_t type, uint8_t *data, size_t length) {
+  switch (type) {
+    case (WStype_CONNECTED):
+      Serial.printf("Device is COnnected");
+      NodeWSocket.sendTXT("{\"msgType\":\"auth\",\"userType\":\"HardWare\",\"username\":\"NodeMCU\"}");
+      break;
+
+    case (WStype_TEXT):
+      deserializeJson(doc, data);
+      JsonObject message = doc.as<JsonObject>();
+      //int led = message["led"].as<int>();
+      int led = message[String("led")];
+      int outPutN = led + 4;
+      digitalWrite(outPutN, !digitalRead(outPutN));
+      String switchState;
+      if (digitalRead(outPutN) == HIGH) {
+        switchState = "on";
+      }
+      else {
+        switchState = "off";
+      }
+      NodeWSocket.sendTXT("{\"msgType\":\"control\",\"led\":\"outPutN\",\"className\":\"switchState\"}");
```

Comment:
This check is unnecessary; You only need to send the number/ index of the led, and the [frontend code will just flip it's state](https://github.com/mhashim6/nodejs-iot/blob/711355ebe41fd87965bc6f8168a385e39c8c8324/frontend/script.js#L49). It needs not to know about the specific state of the led, it only needs to know if it's being flipped. You should replace them with:
```
      char *msg;
      sprintf(msg,"{\"msgType\": \"control\",\"led\":\"%d\"}", led);
      NodeWSocket.sendTXT(msg);
```

Reply:
Exactly.